pkg/pillar/cmd/downloader/downloader.go

```diff
@@ -1529,15 +1559,20 @@ func rktCreateAuthFile(config *types.DownloaderConfig,
 
 func rktFetchContainerImage(ctx *downloaderContext, key string,
 	config types.DownloaderConfig, status *types.DownloaderStatus,
-	dsCtx types.DatastoreContext) error {
+	dsCtx types.DatastoreContext, pullPolicy string) error {
 	// update status to DOWNLOAD STARTED
 	status.State = types.DOWNLOAD_STARTED
 	publishDownloaderStatus(ctx, status)
 
 	imageID := ""
+	var authFile string
+	registry, downloadURL, err := getRegistry(dsCtx.DownloadURL)
+	if err != nil {
+		log.Errorf("getRegistry: registry and download url parsing failed. err: %+v", err)
```

Comment:
No point in moving forward here.. Return this error. This should be propagated all the way to cloud.